packages/block-editor/src/store/selectors.js

```diff
@@ -1419,6 +1419,23 @@ export function canMoveBlocks( state, clientIds, rootClientId = null ) {
 	);
 }
 
+/**
+ * Determines if the given block type can be locked/unlocked by a user.
+ *
+ * @param {Object}          state      Editor state.
+ * @param {(string|Object)} nameOrType Block name or type object.
+ *
+ * @return {boolean} Whether a given block type can be locked/unlocked.
+ */
+export function canLockBlockType( state, nameOrType ) {
+	if ( ! hasBlockSupport( nameOrType, 'lock', true ) ) {
```

Comment:
I went with a simple name for the flag, but happy to change it if necessary.

Reply:
One of the confusing aspects about this is that supporting `lock` here might means two things:
- A block can be locked and unlocked (via attributes and UI).
- The UI to change locking is disabled.

I know that all blocks support locking, and this support flag is just for the UI, it feels a bit confusing but `lockUI` doesn't look as clean.

Reply:
I agree; the support flag name doesn't really represent what the setting might do.

Maybe we should use `__experimentalLock` while thinking about the permanent name.

@mtias What do you think?

Reply:
I think it's okay to go with `lock` as long as [documentation](https://github.com/WordPress/gutenberg/pull/39568/files#diff-cb2052b95e3c5b5fa711f6519d1e3c376597cd1747e6a0a781dc51447255ef16) explains what this flag does.

Reply:
Let's add it as experimental for a release or two but aim at formalizing it before 6.0 is released.

Reply:
Updated in [515c3e7](https://github.com/WordPress/gutenberg/pull/39568/commits/515c3e7db7823dc94855654650f1f00afe39e96b) and created todo issue for stabilization - #39809.